src/workerd/io/observer.h

```diff
@@ -45,6 +57,12 @@ class RequestObserver: public kj::Refcounted {
     return kj::none;
   };
 
+  // This is called when a writable byte stream is created whilst processing this request. It will
+  // be destroyed when the corresponding byte stream is destroyed.
+  virtual kj::Maybe<kj::Own<ByteStreamObserver>> tryCreateWritableByteStreamObserver() {
+    return kj::none;
+  }
```

Comment:
Might be a bit easier to just have this return a `kj::Own<ByteStreamObserver>` that wraps a non-op singleton instance so that we can skip the `KJ_IF_SOME` block.

Reply:
Good idea. It'd make sense to do this for both `ByteStreamObserver` and `WebSocketObserver` so I'll do both in a follow up PR since changing `tryCreateWebSocketObserver` will be a breaking API change.